app/assets/javascripts/discourse/app/components/sidebar/anonymous-categories-section.js

```diff
@@ -0,0 +1,45 @@
+import { cached } from "@glimmer/tracking";
+import { inject as service } from "@ember/service";
+import I18n from "I18n";
+import Component from "@glimmer/component";
+import CategorySectionLink from "discourse/lib/sidebar/categories-section/category-section-link";
+
+export default class SidebarAnonymousCategoriesSection extends Component {
+  @service topicTrackingState;
+  @service site;
+  @service siteSettings;
+
+  @cached
+  get sectionLinks() {
+    let categories = this.site.categories;
+
+    if (this.siteSettings.default_sidebar_categories) {
+      const defaultCategoryIds = this.siteSettings.default_sidebar_categories
+        .split("|")
+        .map((categoryId) => parseInt(categoryId, 10));
+
+      categories = categories.filter((category) =>
+        defaultCategoryIds.includes(category.id)
+      );
+    } else {
+      categories = categories
```

Comment:
Just confirming this is already sorted by the top categories right?

Reply:
Actually, I had to change it to site.categoriesList, but you are right. It returns sorted categories (unless fixed positions - then they are returned in fixed positions order)